dissect/target/filesystems/zip.py

```diff
@@ -0,0 +1,136 @@
+import io
+import logging
+import zipfile
+
+from dissect.util.stream import BufferedStream
+
+from dissect.target.exceptions import FileNotFoundError, FilesystemError
+from dissect.target.filesystem import Filesystem, VirtualFile, VirtualFilesystem
+from dissect.target.helpers import fsutil
+
+log = logging.getLogger(__name__)
+
+
+class ZipFilesystem(Filesystem):
+    """This Zip filesystem implementation is WIP and
+    has only been tested for opening java jar archives.
+    Its structure has been copied from the tar filesystem
+    implementation.
+    """
+
+    __fstype__ = "zip"
+
+    def __init__(self, fh, base=None, case_sensitive=True, zipinfo=None, *args, **kwargs):
```

Comment:
```suggestion
    def __init__(self, fh: BinaryIO, base: Optional[str]=None, alt_separator: str = "", case_sensitive: bool = True, *args, **kwargs):
```
Remove `zipinfo` (it's not used), add `alt_separator` and add types.